openmc/mgxs/mgxs.py

```diff
@@ -10,6 +10,11 @@
 
 import numpy as np
 
+# Require numpy to print output in scientific notation to 6 decimal places.
+# This is needed to avoid round off error when large numbers are printed,
+# which can cause tests to fail for different build configurations.
+np.set_printoptions(formatter={'float': lambda x: format(x, '8.6E')})
```

Comment:
When the velocities were printed to file in the `test_mgxs_library_hdf5`, they were printed to ~16 digits of precision which caused different build configurations to disagree on the last 2-3 digits. This line requires numpy to always print out numbers in scientific notation to six decimal places. I'll be the first to admit this is kind of a hack. Nevertheless it is an issue that will come back if we just make a special case for printing the velocities where the values are truncated after a certain number of decimal places. @paulromano @wbinventor and others - if you have any thoughts on a more elegant (and preferably general) solution, I'm all ears.